ffi-ctests/ctests/tests.c

```diff
@@ -0,0 +1,270 @@
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include <wirefilter.h>
+
+#define WIREFILTER_STRING(v, s) {(v).data = (const unsigned char *)(s); (v).length = strlen(s);}
+
+bool test_wirefilter_ffi_01() {
+	wirefilter_Scheme *scheme = wirefilter_create_scheme();
+	if(!scheme) {
+		return false;
+	}
+	wirefilter_free_scheme(scheme);
+	return true;
+}
+
+bool test_wirefilter_ffi_02() {
+	wirefilter_Scheme *scheme = wirefilter_create_scheme();
+	if(!scheme) {
+		return false;
+	}
+
+	wirefilter_ExternallyAllocatedStr field;
+
+	WIREFILTER_STRING(field, "http.host");
+	wirefilter_add_type_field_to_scheme(scheme, field, wirefilter_Bytes);
+	WIREFILTER_STRING(field, "ip.addr");
+	wirefilter_add_type_field_to_scheme(scheme, field, wirefilter_Ip);
+	WIREFILTER_STRING(field, "ssl");
+	wirefilter_add_type_field_to_scheme(scheme, field, wirefilter_Bool);
+	WIREFILTER_STRING(field, "tcp.port");
+	wirefilter_add_type_field_to_scheme(scheme, field, wirefilter_Int);
+
+	wirefilter_free_scheme(scheme);
+	return true;
+}
+
+bool test_wirefilter_ffi_03() {
```

Comment:
Instead of `bool`, maybe use `assert` in each place? That would give you clear error messages and line numbers on failures.

Reply:
Yep I fully agree.